Our code has an incompatiblity with updated versions of @octokit/rest from 18.1.0 to 18.3.0

We need to fix our code so we can take updates for: @octokit/rest from 18.1.0 to 18.3.0
The error we have is:
```
 [6:36:30 PM] x src/createGithubReleases.ts(95,5): error TS2769: No overload matches this call.
ERR!   The last overload gave the following error.
ERR!     Argument of type '{ (params?: (RequestParameters & Pick<{ owner: string; repo: string; } & { per_page?: number | undefined; page?: number | undefined; }, "per_page" | "page" | "owner" | "repo">) | undefined): Promise<...>; defaults: <O extends RequestParameters = RequestParameters>(newDefaults: O) => RequestInterface<...>; endpoint: ...' is not assignable to parameter of type 'RequestInterface<object>'.
ERR!       The types of 'endpoint.DEFAULTS' are incompatible between these types.
ERR!         Type '{ url: string; } & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string | undefined; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type 'object & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string | undefined; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }'.
ERR!           Type '{ url: string; } & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string | undefined; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type 'RequestParameters'.
ERR!             Types of property 'request' are incompatible.
ERR!               Type 'import("D:/a/_work/1/s/node_modules/@octokit/plugin-rest-endpoint-methods/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions | undefined' is not assignable to type 'import("D:/a/_work/1/s/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions | undefined'.
ERR!                 Type 'import("D:/a/_work/1/s/node_modules/@octokit/plugin-rest-endpoint-methods/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions' is not assignable to type 'import("D:/a/_work/1/s/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions'.
ERR!                   Types of property 'agent' are incompatible.
ERR!                     Type 'unknown' is not assignable to type 'Agent | undefined'.
ERR!                       Type 'unknown' is not assignable to type 'Agent'.
ERR! src/createGithubReleases.ts(100,42): error TS2571: Object is of type 'unknown'.
ERR! src/createGithubReleases.ts(106,44): error TS2345: Argument of type 'PaginationResults<unknown>' is not assignable to parameter of type '{ tag_name: string; }[]'.
ERR!   Type 'unknown' is not assignable to type '{ tag_name: string; }'.
```

For now updates to this pacakge have been postponed. See #7236

--- a/change/@rnw-scripts-create-github-releases-77b86566-6968-4cb6-b44a-87e691a48ea7.json
+++ b/change/@rnw-scripts-create-github-releases-77b86566-6968-4cb6-b44a-87e691a48ea7.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Update @octokit/rest",
+  "packageName": "@rnw-scripts/create-github-releases",
+  "email": "ngerlem@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/@rnw-scripts/create-github-releases/package.json
+++ b/packages/@rnw-scripts/create-github-releases/package.json
@@ -13,7 +13,7 @@
     "create-github-releases": "./bin.js"
   },
   "dependencies": {
-    "@octokit/rest": "^18.0.12",
+    "@octokit/rest": "^18.5.3",
     "@react-native-windows/find-repo-root": "^0.0.0-canary.19",
     "chalk": "^4.1.0",
     "glob": "^7.1.6",

--- a/packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts
+++ b/packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts
@@ -92,7 +92,7 @@ const octokit = new Octokit({
 
   console.log('Fetching releases...');
   const githubReleases = await octokit.paginate(
-    octokit.repos.listReleases,
+    'GET /repos/{owner}/{repo}/releases',
     RNW_REPO,
   );
 

